pkgs/development/compilers/cakelisp/default.nix

```
@@ -0,0 +1,41 @@
+{ stdenv, fetchFromGitHub, gcc }:
+
+stdenv.mkDerivation {
+  pname = "cakelisp";
+  version = "unstable-2020-12-23";
+
+  src = fetchFromGitHub {
+    owner = "makuto";
+    repo = "cakelisp";
+    rev = "1754288817ea325b466a6b80d2c29beff6f5580a";
+    sha256 = "1l7vg4y3ysxdjk3hn8mbhcnw3k3lnpa268j39k8m0m7b45p6lgwa";
+  };
+
+  buildInputs = [ gcc ];
+
+  postPatch = ''
+
+    ${stdenv.lib.optionalString stdenv.isDarwin "substituteInPlace Build.sh --replace '--export-dynamic' '-export_dynamic'"}
+    ${stdenv.lib.optionalString stdenv.isDarwin "substituteInPlace runtime/HotReloading.cake --replace '--export-dynamic' '-export_dynamic'"}
+    ${stdenv.lib.optionalString stdenv.isDarwin "substituteInPlace Bootstrap.cake --replace '--export-dynamic' '-export_dynamic'"}
+
+    substituteInPlace runtime/HotReloading.cake \
+        --replace '"/usr/bin/g++"' '"${gcc}/bin/g++"'
+    substituteInPlace src/ModuleManager.cpp \
+        --replace '"/usr/bin/g++"' '"${gcc}/bin/g++"'
+  '';
```

```suggestion
    substituteInPlace runtime/HotReloading.cake \
        --replace '"/usr/bin/g++"' '"${gcc}/bin/g++"'
    substituteInPlace src/ModuleManager.cpp \
        --replace '"/usr/bin/g++"' '"${gcc}/bin/g++"'
  '' + stdenv.lib.optionalString stdenv.isDarwin ''
    substituteInPlace Build.sh --replace '--export-dynamic' '-export_dynamic'
    substituteInPlace runtime/HotReloading.cake --replace '--export-dynamic' '-export_dynamic'
    substituteInPlace Bootstrap.cake --replace '--export-dynamic' '-export_dynamic'
    '';
```

Didn't test but repeating optionalString three times is a bit repetitive.